airbyte-integrations/connectors/source-apple-search-ads/source_apple_search_ads/apple_search_ads.yaml

```diff
@@ -0,0 +1,228 @@
+      client_id: "{{ config['client_id'] }}"
+      client_secret: "{{ config['client_secret'] }}"
+      refresh_token: ""
+    request_options_provider:
+      request_headers:
+        X-AP-Context: "orgId={{ config['org_id'] }}"
+  retriever:
+    requester:
+      $ref: "*ref(definitions.requester)"
+      error_handler:
+        response_filters:
+          - http_codes: [ 500 ]
+            action: RETRY
+
+  base_stream:
+    selector:
+      extractor:
+        field_pointer: [ "data" ]
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+      record_selector:
+        $ref: "*ref(definitions.base_stream.selector)"
+      paginator:
+        type: DefaultPaginator
+        pagination_strategy:
+          type: OffsetIncrement
+          page_size: 1000
+        page_size_option:
+          inject_into: request_parameter
+          field_name: limit
+        page_token_option:
+          field_name: offset
+          inject_into: request_parameter
+        url_base: "*ref(definitions.requester.url_base)"
+
+  campaigns_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "campaigns"
+      primary_key: "id"
+      path: "/campaigns"
+
+  stream_slicer_campaign:
+    type: SubstreamSlicer
+    parent_stream_configs:
+      - stream: "*ref(definitions.campaigns_stream)"
+        parent_key: "id"
+        stream_slice_field: "campaign_id"
+
+  adgroups_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "adgroups"
+      primary_key: "id"
+    retriever:
+      $ref: "*ref(definitions.base_stream.retriever)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        path: "/campaigns/{{ stream_slice.campaign_id }}/adgroups"
+      stream_slicer:
+        type: CartesianProductStreamSlicer
+        stream_slicers:
+          - "*ref(definitions.stream_slicer_campaign)"
+
+  stream_slicer_adgroup:
+    type: SubstreamSlicer
+    parent_stream_configs:
+      - stream: "*ref(definitions.adgroups_stream)"
+        parent_key: "id"
+        stream_slice_field: "adgroup_id"
+
+  keywords_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "keywords"
+      primary_key: "id"
+    retriever:
+      $ref: "*ref(definitions.base_stream.retriever)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        path: "/campaigns/{{ stream_slice.parent_slice.campaign_id }}/adgroups/{{ stream_slice.adgroup_id }}/targetingkeywords"
+      stream_slicer:
+        type: CartesianProductStreamSlicer
+        stream_slicers:
+          - "*ref(definitions.stream_slicer_adgroup)"
+
+  stream_slicer_date:
+    type: "DatetimeStreamSlicer"
+    start_datetime:
+      datetime: "{{ config['start_date'] }}"
+      datetime_format: "%Y-%m-%d"
+    end_datetime:
+      datetime: "{{ config['end_date'] or today_utc() }}"
+      datetime_format: "%Y-%m-%d"
+    step: "1d"
+    lookback_window: "30d"
```

Comment:
The `lookback_window` is critical here because this API fetches marketing data with an [attribution window of 30 days](https://developer.apple.com/documentation/iad/setting_up_apple_search_ads_attribution), ie any data < 30 days needs to be refreshed with an Incremental Dedup sync.

However, the `lookback_window` is currently bugged as it is applied on the `start_date` instead of the `cursor_field`: issue #15628

Reply:
@girarda any ideas if it's possible to solve #15628

Reply:
We don't have an ETA on this issue yet.

@rach-r https://github.com/airbytehq/airbyte/issues/15628 is a good first issue if this is blocking you and you're interested in contributing. Your contribution would be more than welcome!

Reply:
@girarda I can see the issue is solved now, fixing the connector in Incremental Dedup mode. Thank you and @maxi297 for your help 🙏